packages/lit-dev-content/site/docs/ssr/authoring.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Authoring components for Lit SSR
+eleventyNavigation:
+  key: Authoring components
+  parent: Server rendering
+  order: 4
+---
+
+{% labs-disclaimer %}
+
+Lit's approach to rendering web components in a server environment places some restrictions on component code to achieve efficient server rendering. When authoring components, keep in mind these considerations to ensure they are compatible with Lit SSR.
+
+Note: The restrictions listed on this page are subject to change as we make improvements to Lit SSR. If you would like to see a certain use case supported, please [file an issue](https://github.com/lit/lit/issues/new/choose) or start a [discussion](https://github.com/lit/lit/discussions) thread.
+
+## Considerations
+
+### Browser only code
+
+Most browser DOM APIs are not available in the Node environment. Lit SSR utilizes a DOM shim that's the bare minimum required for rendering Lit templates and components. For a full list of what APIs are available, see the [DOM Emulation](/docs/ssr/dom-emulation) page.
+
+When authoring components, perform imperative DOM operations from lifecycle methods that are called only on the client, and not on the server. See the [lifecycle](#lifecycle) section below for what specific methods can be used.
+
+Some component modules may also have side effects that utilize browser APIs, for example to detect certain browser features, which would break simply by importing in a non-browser environment. If these side-effects cannot be moved to a client-only lifecycle, you must provide proper alternatives or some default behavior for when running in Node.
```

Comment:
This paragraph confused me. There's a lot packed in here, plus several pretty vague references (like "proper alternatives").

I think you're saying, roughly: "You might have a module that defines a Lit component, but _also_ has a side effect that could break on import. In this case, you have a few choices:

- Move the side effect code into a Lit lifecycle callback that's only called on the browser.
- Conditionalize the side-effect code such that it doesn't run on the server.
- I'm actually not sure what else "property alternatives or some default behavior" might involve."

If so, here's my first stab at a rephrase.

```suggestion
Some modules that define Lit components may also have side effects that use browser APIs—for example to detect certain browser features—such that the module breaks when imported in a non-browser environment. In this case, you can move the side effect code into a browser-only lifecycle callback, or conditionalize so that it only runs on the browser.
```

Also given that the section is called "Browser-only code," consider standardizing on "browser-only" (like, "browser-only callback" maybe) instead of switching between "browser" and "client" which can make things a little less clear.